docs/java-rest/high-level/apis/search.asciidoc

```diff
@@ -1,4 +1,119 @@
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-request-indicesOptions]
+--------------------------------------------------
+
+The preference parameter can e.g. be used to execute the search to prefer local shards. This defaults to randomize across shards.
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-request-preference]
+--------------------------------------------------
+
+==== Using the SearchSourceBuilder
+
+Most options controlling the search behavior can be set on the `SearchSourceBuilder`,
+which contains more or less the equivalent of the options in the search request body of the Rest API.
+
+Here are a few examples of some common options:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-source-basics]
+--------------------------------------------------
+<1> Create a `SearchSourceBuilder` with default options.
+<2> Set the query. Can be any type of `QueryBuilder`
+<3> Set the `from` option that determines the result index to start searching from. Defaults to 0.
+<4> Set the `size` option that determines the number of search hits to return. Defaults to 10.
+<5> Set an optional timeout that controls how long the search is allowed to take.
+
+==== SearchResponse
+
+When the search is executed, it returns a `SearchResponse` object that serves
+as an entry point to a lot of information about the search response and the
+results:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-response-basic]
+--------------------------------------------------
+
+<1> Actually perform the search. The response contains information about the
+execution of the request itself, like the `took` time or if the request timed
+out.
+<2> Get the `SearchHits` from the response. This object provides access to
+global result parameters like number of total hits or the maximum score.
+<3> Get the individual `SearchHit` objects. These are returned in the order
+determined by the search. `SearchHit` provides access to document details like
+its id or the document `source`.
+
+==== Accessing SearchResponse details
+
+The `SearchResponse` that is returned by performing `client.search(searchRequest)` provides details about the search execution as well as access to the document returned as a result. First, lets look at some of the information provided by the `SearchResponse` itself:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
```

Comment:
what is the difference between this example and the previous one? I would maybe merge the two.